src/display.c

```diff
@@ -136,25 +136,39 @@ static void SurfaceRecord_DumpRects(SurfaceRecord record,
 static size_t LCUIDisplay_RenderSurface(SurfaceRecord record)
 {
 	size_t count = 0;
+	int i;
 	LCUI_Rect *rect;
 	LCUI_PaintContext paint;
 	LCUI_SysEventRec ev;
 	LinkedList rects;
 	LinkedListNode *node;
 	LCUI_BOOL can_render;
+	LCUI_Rect **rectArray;
 
 	ev.type = LCUI_PAINT;
 	can_render = record->widget && record->surface &&
 		     Surface_IsReady(record->surface);
 	record->rendered = FALSE;
 	LinkedList_Init(&rects);
 	SurfaceRecord_DumpRects(record, &rects);
+
+	rectArray = (LCUI_Rect **)malloc(sizeof(LCUI_Rect*) * rects.length);
+	i = 0;
+	for (LinkedList_Each(node, &rects)) {
```

Comment:
Earlier I was curious how OpenMP would traverse the nodes in the linked list. Do your changes mean that the array is more suitable for OpenMP?

Reply:
Yes, it is.
`LinkedList_Each()` macro's loop form does not follow "Canonical Loop Form" defined in OpenMP.
I rewritten the loop form to "Canonical Loop Form" by converting the `rects` linked list to an array.
This is a change to address the error in the previous review (https://github.com/lc-soft/LCUI/pull/189#pullrequestreview-332215928) and is not related to performance.
Would an you try compiling with vs2017?

Reply:
I added some debug message output, OpenMP does work.

```
rects: 4
rect: (0,0,800,450), thread: 0
rect: (800,0,800,450), thread: 1
rect: (0,450,800,450), thread: 2
rect: (800,450,800,450), thread: 3
count: 3609
rects: 4
rect: (800,0,800,450), thread: 1
rect: (0,0,800,450), thread: 0
rect: (0,450,800,450), thread: 2
rect: (800,450,800,450), thread: 3
count: 3609
rects: 4
rect: (0,0,800,450), thread: 0
rect: (800,450,800,450), thread: 3
rect: (800,0,800,450), thread: 1
rect: (0,450,800,450), thread: 2
count: 3609
rects: 4
rect: (0,0,800,450), thread: 0
rect: (800,450,800,450), thread: 3
rect: (800,0,800,450), thread: 1
rect: (0,450,800,450), thread: 2
count: 3609
```

Maybe some code in the renderer is affecting performance, I will debug the performance of the renderer.

Reply:
Thank you.